### ISSUE TITLE
use collected article headlines and fine-tune personal BERT model

### ISSUE THREAD

**Issue by Shubxam** [Shubxam/Nifty-500-Live-Sentiment-Analysis#82]
- have the article_data table with headlines and inferred sentiment labels.
- review (by some means) sentiment labels to make sure they are right.
- take a good base BERT model and finetune on a local headlines.

**Comment by Shubxam** [comment]
tutorial: https://blog.jetbrains.com/pycharm/2024/12/how-to-do-sentiment-analysis-with-large-language-models/#fine-tune-your-own-llm-for-sentiment-analysis

**Comment by Shubxam** [comment]
You can use Encoder only model (which is the architecture of BERT models) and train it for downstream task (in our case, sentiment analysis).

- there is a newer BERT architecture (Called [ModernBERT](https://huggingface.co/blog/modernbert)) which is even better then base BERT models.
- Could use a library like [infinity](https://github.com/michaelfeil/infinity) and [VLLM](https://github.com/vllm-project/vllm) [which now supports ModernBERT](https://github.com/vllm-project/vllm/pull/16648) for high throughput inference.
- ModernBERT usually shows better performance where long context is needed (such as Long News Articles Text). For only classifying headlines, its performance should be similar to BERT.

**Comment by Shubxam** [comment]
Datasets:
- https://www.researchgate.net/publication/251231364_FinancialPhraseBank-v10
- https://huggingface.co/datasets/takala/financial_phrasebank
